src/backend/utils/adt/json.c

```diff
@@ -338,12 +354,67 @@ datum_to_json(Datum val, bool is_null, StringInfo result,
 			break;
 		default:
 			outputstr = OidOutputFunctionCall(outfuncoid, val);
+			// The outfuncoid are specific to babelfish and therefore no macro is defined.
+			if ((sql_dialect==SQL_DIALECT_TSQL) && 
+				(outfuncoid==17325 ||
+				 outfuncoid==17430 ||
+				 outfuncoid==17504 ||
+				 outfuncoid==17639))
+			{
+				StringInfo format_output = makeStringInfo();
+				for_json_path_date_time_format(format_output,outputstr, outfuncoid);
+				outputstr = format_output->data;
```

Comment:
1. `outputstr` returned from line#356 could be leaked.
2. `format_output->data` could be double-freed because of line#369 and freeing from `StringInfo`.